edgesite/pkg/controller/config/kube.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"time"
+
+	"github.com/kubeedge/beehive/pkg/common/config"
+	"github.com/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/edgesite/pkg/controller/constants"
+)
+
+// KubeMaster is the url of edge master(kube api server)
+var KubeMaster string
+
+// KubeConfig is the config used connect to edge master
+var KubeConfig string
+
+// KubeNamespace is the namespace to watch(default is NamespaceAll)
+var KubeNamespace string
+
+// KubeContentType is the content type communicate with edge master(default is "application/vnd.kubernetes.protobuf")
+var KubeContentType string
+
+// KubeQPS is the QPS communicate with edge master(default is 1024)
+var KubeQPS float32
+
+// KubeBurst default is 10
+var KubeBurst int
+
+// NodeID for the current node
+var KubeNodeID string
+
+// NodeName for the current node
+var KubeNodeName string
+
+// KubeUpdateNodeFrequency is the time duration for update node status(default is 20s)
+var KubeUpdateNodeFrequency time.Duration
+
+func init() {
```

Comment:
Alternative to copy field by field, I'd suggest:
1. wrap all `KubeX` fields into a struct
2. wrap all default values into several structs
3. use struct copy / deep copy to set default values for `KubeX` fields, and then rewrite those user indicated ones.

Reply:
Agree. This is true for edgecontroller. Let me raise a code clean up issue.

Reply:
Sure, please link the clean up issue to this PR, and we can get this one accepted.

Reply:
Sounds good. #525 is raised for this purpose.